### PR TITLE
#285: Create an adaptor for CLI Argument parsing and Configuration File Parsing

### PR DESCRIPTION
**Pull request by adityatrivedi**
Addresses #285.

### CODE REVIEW

**Comment by a4sriniv** [comment]
:+1:

File: src/main/java/com/sleekbyte/tailor/common/ConfigProperties.java [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import java.util.Properties;
 
 /**
- * Configuration Properties container.
+ * YamlConfiguration Properties container.
```
Comment (alykhank) [inline src/main/java/com/sleekbyte/tailor/common/ConfigProperties.java]: This should not be renamed, it is about the `config.properties` file.

File: src/test/java/com/sleekbyte/tailor/functional/TailorYamlConfigurationTest.java [hunk 2]
```diff
@@ -33,7 +33,7 @@
  * Create config file that will only lint UpperCamelCaseTest.swift functional test file.
  */
 @RunWith(MockitoJUnitRunner.class)
-public final class TailorConfigurationTest {
+public final class TailorYamlConfigurationTest {
```
Comment (alykhank) [inline src/test/java/com/sleekbyte/tailor/functional/TailorYamlConfigurationTest.java]: Can we actually rename this file and class to `YamlConfigurationTest` instead?

File: src/main/java/com/sleekbyte/tailor/common/ConfigProperties.java (hunk 1 above)
Comment (adityatrivedi) [inline src/main/java/com/sleekbyte/tailor/common/ConfigProperties.java]: Umm what should the adapter class be named then?

Comment (adityatrivedi) [inline src/main/java/com/sleekbyte/tailor/common/ConfigProperties.java]: YamlConfiguration makes it explicitly clear that the class is related to .tailor.yml file.

**Comment by alykhank** [comment]
:+1:

**Pull request merged**